test-suite/djinni/foo_client_interface.djinni

```diff
@@ -0,0 +1,30 @@
+foo_extensible_record = record +p +c {
+    number1: i32;
+    string1: string;
+}
+
+foo_some_other_record = record {
+    number1: i64;
+    number2: i32;
+} deriving (eq, ord)
+
+foo_client_returned_record = record {
+    record_id: i64;
+    content: string;
+
+    some_record: foo_some_other_record;
+}  deriving (eq, ord)
+
+foo_client_interface = interface +c {
+    # Returns record of given string
+    set_record(rec: foo_client_returned_record);
+    get_record(): foo_client_returned_record;
+
+    set_extensible_record(rec: foo_extensible_record);
+    get_extensible_record(): foo_extensible_record;
+
+    get_extensible_record_number2(): i32;
+    get_extensible_record_string2(): string;
+
+    static create(): foo_client_interface;
+}
```

Comment:
I think the concept of having separate test interfaces for djinni in the same folder is kind of confusing. Do you think the tests could be merged with the default tests for other languages to some extend? 

As a starter maybe we could move them in a folder labeled as python-specific or rename them to something more specific, like `python_client_interface.djinni` instead of `foo_...`?

Reply:
I definitely want to do that! IMO, right now this PR is in a "mergeable" state. I'd like to incrementally include the preexisting tests to Python and remove the "foo" ones; that will take making sure all Djinni features are supported by the Python generator, including proxying.

Reply:
If it is in a "mergeable" state we should possible merge it  t be able to move on with such changes, and it becomes more easy to have a state to branch from to try out different things, and work towards  a "releasable" state.

I will therefore approve now, but give it  a sort moment before merge to check if @jothepro would like to run a different strategy.

Reply:
Nope, I'm fine with that! Just wanted to mention that I noticed this 👍. I'm going to give my final approval asap :)